bucket.go

```diff
@@ -203,7 +202,7 @@ func OpenBucket(urlStr string, bucketName string, mode OpenMode) (b *Bucket, err
 	}
 
 	registerBucket(bucket)
-	return bucket, err
+	return bucket.copy(), err
```

Comment:
What happens in the case of concurrent OpenBucket requests for a bucket that doesn't already exist in the registry?  It doesn't look like we're doing any locking, so in the case that registerBucket finds an existing bucket in the registry at line 205 (i.e. this instance of OpenBucket lost the race), I think we should be returning a copy of that bucket, and not the newly created one.